test/conformance/walk.go

```diff
@@ -14,415 +14,294 @@ See the License for the specific language governing permissions and
+	// with the ConformanceIt(...) call and the next call in that callstack will be the
+	// ast.Node which is attached to the comment that we want.
+	for i := len(spec.ComponentCodeLocations) - 1; i > 0; i-- {
+		fullstacktrace := spec.ComponentCodeLocations[i].FullStackTrace
+		c, err = getConformanceDataFromStackTrace(fullstacktrace)
+		if err != nil {
+			log.Printf("Error looking for conformance data: %v", err)
 		}
-		if len(call.Args) < 1 {
-			v.failf(call, "Not enough arguments to It()")
+		if c != nil {
+			break
 		}
-	default:
-		return false
 	}
 
-	switch arg := call.Args[0].(type) {
-	case *ast.BasicLit:
-		if arg.Kind != token.STRING {
-			v.failf(arg, "Unexpected non-string argument to It()")
-		}
-		if strings.Contains(arg.Value, "[Conformance]") {
-			return true
-		}
-	default:
-		// non-literal argument to It()... we just ignore these even though they could be a way to "sneak in" a conformance test
+	if c == nil {
+		log.Printf("Did not find test info for spec: %#v\n", getTestName(spec))
+		return nil
 	}
 
-	return false
+	c.CodeName = getTestName(spec)
+	return c
 }
 
-func (v *visitor) failf(expr ast.Expr, format string, a ...interface{}) {
-	msg := fmt.Sprintf(format, a...)
-	fmt.Fprintf(os.Stderr, "ERROR at %v: %s\n", v.FileSet.Position(expr.Pos()), msg)
-}
-
-func (v *visitor) comment(x *ast.BasicLit) string {
-	for _, comm := range v.cMap.Comments() {
-		testOffset := int(x.Pos()-comm.End()) - len("framework.ConformanceIt(\"")
-		//Cannot assume the offset is within three or four tabs from the test block itself.
-		//It is better to trim the newlines, tabs, etc and then we if the comment is followed
-		//by the test block itself so that we can associate the comment with it properly.
-		if 0 <= testOffset && testOffset <= 10 {
-			b1 := make([]byte, x.Pos()-comm.End())
-			//if we fail to open the file to compare the content we just assume the
-			//proximity of the comment and apply it.
-			myf, err := os.Open(v.FileSet.File(x.Pos()).Name())
-			if err == nil {
-				defer myf.Close()
-				if _, err := myf.Seek(int64(comm.End()), 0); err == nil {
-					if _, err := myf.Read(b1); err == nil {
-						if strings.Compare(strings.Trim(string(b1), "\t \r\n"), "framework.ConformanceIt(\"") == 0 {
-							return comm.Text()
-						}
-					}
-				}
-			} else {
-				//comment section's end is noticed within 10 characters from framework.ConformanceIt block
-				return comm.Text()
-			}
-		}
-	}
-	return ""
+func getTestName(spec *types.SpecSummary) string {
+	return strings.Join(spec.ComponentTexts[1:], " ")
 }
 
-func (v *visitor) emit(arg ast.Expr) {
-	switch at := arg.(type) {
-	case *ast.BasicLit:
-		if at.Kind != token.STRING {
-			v.failf(at, "framework.ConformanceIt() called with non-string argument")
-			return
-		}
-
-		description := v.getDescription(at.Value)
-		err := validateTestName(description)
+// TODO(SCHNAKE) 1-2 tests for each case would be good
```

Comment:
Remove todo; have to double check if I made the intended tests or not (its been a little while since I looked at this; I think I was done though)